Support metric_base_uri for newrelic metrics provider

knapo opened the pull request:
Statuspage API requires to provide `metric_base_uri` param to create NewRelic-type `metrics_providers`.

yannh commented:
Thx! The SDK patch needs to be made against this repo https://github.com/yannh/statuspage-go-sdk/blob/master/metrics_provider.go , it is only in this repo as a vendored dependency :) I was wondering if it was possible to use a ValidateFunc ( see https://www.terraform.io/docs/extend/schemas/schema-behaviors.html ) to ensure the parameter is set if the metric provider is newrelic but it seems the validation function can not access the rest of the parameters :thinking: 
Thanks a lot for the contribution, it needs a tiny bit of work, I'll get back to you soon!

yannh commented:
@knapo I can't see this parameter documented in their API https://developer.statuspage.io/#operation/postPagesPageIdMetricsProviders - could you point to where you found it?

knapo commented:
@yannh Indeed, it doesn't seem to be there. I contacted the support, however it's in `GET` metrics_provider payload and  I'm getting `Metric base uri can't be blank` error when it's not provided on create.

knapo commented:
Response from statuspage support:

> We have added the "metric_base_uri" field to the New Relic integration only, and updated our docs page reflect that; however, it's on our New Relic documentation, not on the API page.
>
> https://support.atlassian.com/statuspage/docs/add-a-new-relic-metric/#Add-a-New-Relic-metric

yannh commented:
Interesting.. Did you actually get it to work? The API used by their WebUI is similar but subtly different to the public API :D I'll give it a shot soon :)

knapo commented:
Yes. POST to `https://api.statuspage.io/v1/pages/..../metrics_providers` created a resource. Also, `metric_base_uri` is available when I get the resource: `https://api.statuspage.io/v1/pages/.../metrics_providers/...`
```
{
    "created_at": "2020-06-22T08:47:57.138Z",
    "updated_at": "2020-06-22T08:47:57.138Z",
    "last_revalidated_at": "2020-06-25T20:03:37.876Z",
    "disabled": false,
    "metric_base_uri": "https://api.newrelic.com/api/v1",
    "id": "....",
    "page_id": "...",
    "type": "NewRelic"
}
```

davidmontoyago commented:
> @knapo I can't see this parameter documented in their API docs https://developer.statuspage.io/#operation/postPagesPageIdMetricsProviders - could you point to where you found it?

Hi, the attribute has just been added to the statuspage.io API docs. I had also reached out to statuspage support for the same thing while working on setting up a Datadog metrics provider.

davidmontoyago commented:
Can we also specify the attribute is required for Datadog as well a NewRelic? I can confirm these are required when configuring a Datadog provider.

This is what I was planning on submitting as a PR before @knapo beat me to it 😄 :

https://github.com/davidmontoyago/terraform-provider-statuspage/commit/466610bf16f009bc35ff9962488abe462532ead0
https://github.com/davidmontoyago/statuspage-go-sdk/commit/80b9287e4cacf08ded0098510c380bd0d0ed41eb

abocan-atl commented:
Hey friends, just wanted to drop by and thank you for the work here.

It's immensely helpful to us, so please let us know whenever you find inconsistencies, we'll do our best to get them sorted as soon as possible 🙇
